src/apps/authentication/LoginPage.tsx

```diff
@@ -79,5 +81,10 @@ const mapStateToProps = (state: any) => ({
   errors: authErrorsSelector(state)
 });
 const mapDispatchToProps = { login: loginAction };
+let { from } = this.props.location.state || { from: { pathname: '/' } } // retrieves the URL set on the PrivateRoute component above
+  if (result.payload.data.is_admin) { // checks if user is admin to redirect to admin page
+      from = '/admin';
+   } 
+   this.props.history.push(from); // this will redirect to intended page
```

Comment:
This code needs to be moved into the React component, not here.

In particular, we need to replace the line:

```ts
if (isLoggedIn) return <Redirect to="/" />
```

With something like:

```ts
if (isLoggedIn) {
  if (nextPage) {
    clearNextPage();
    return <Redirect to={nextPage} />;
  }
  return <Redirect to="/" />;
}
```

This will handle updating the location, etc. `nextPage` is clearly a property that needs to be defined and added. It can then be populated it the `AuthenticationRequired` component by changing this line:

```ts
if (!isLoggedIn) return <Redirect to="/login" />;
```

To something slightly more sophisticated, like:

```ts
const location = useLocation();

if (!isLoggedIn) {
  const currentPage = location.pathname + location.search + location.hash;
  setNextPage(currentPage);
  return <Redirect to="/login" />;
}
```

`setNextPage()` and `clearNextPage()` should be defined in `src/apps/authentication/redux/actions.ts` (as these will be actions controlling a `nextPage` property on the redux reducer).

See how far you can get with implementing those.